src/System/Logger/Types.hs

```diff
@@ -284,6 +285,107 @@ popLabel = localScope $ \case { [] → []; (_:t) → t }
 clearScope ∷ MonadLog a m ⇒ m α → m α
 clearScope = localScope $ const []
 
+-- | Instances of 'MonadLog' that allow to obtain a 'LogFunctionIO' as plain
+-- value. This is helpful when dealing with frameworks that take a logging
+-- function in 'IO' as parameter.
+--
+-- An instance of this class should apply the 'LogLevel', 'LogScope', and
+-- 'LogPolicy' at the time when 'logFunIO' is called and not when the returned
+-- action is excecuted. If the returned action is excecuted after the logger
+-- got released or otherwise invalidated the behavior should match the behavior
+-- on a congested logging pipeling accorrding to the log-policy that was in
+-- scope when 'logFunIO' was called.
+--
+-- Even though it can be very convenient, instances of this class must be used
+-- with care. The action may contain in its closure a reference to some
+-- internal state of the 'MonadLog' instance. Beside of being a source of
+-- potential memory leaks, there also is nothing that prevents a programer to
+-- call this action outside of the valid scope of the 'MonadLog' instance. In
+-- case that the context of the 'MonadLog' instance depends on some state that
+-- gets explicitely deallocated this action may have unexectped behavior.
+--
```

Comment:
I know it's not nice. It's a bit like `unsafePerformIO` for a monad different than IO. I wonder if the same purpose could be achieved with an instance of `MonadBaseControl`?

[On the other hand what consider safe in other places is not necessarily safer than what we have here. For instance when `withLogger ... \logger -> ...` is used nothing (but best practices and convention) is preventing a programer from using `logger` outside the scope of `withLogger`.]